src/libcore/cmp.rs

```diff
@@ -443,6 +443,42 @@ pub trait Ord: Eq + PartialOrd<Self> {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     fn cmp(&self, other: &Self) -> Ordering;
+
+    /// Compares and returns the maximum of two values.
+    ///
+    /// Returns the second argument if the comparison determines them to be equal.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(ord_max_min)]
+    ///
+    /// assert_eq!(2, 1.max(2));
+    /// assert_eq!(2, 2.max(2));
+    /// ```
+    #[unstable(feature = "ord_max_min", issue = "25663")]
+    fn max(self, other: Self) -> Self
+    where Self: Sized {
```

Comment:
I agree that `T` is a more helpful return type than `&T`.

Also, https://github.com/rust-lang/rust/issues/25663#issuecomment-267128640 argues for consistency with the inherent impl on fN:

> I often write performance-sensitive graphics code, which I prototype on floats for ease of writing, and then where appropriate change to use integers for speed/memory efficiency. 90% of calculations just work, and max is one of the warts that doesn't.

Shame that `a.min(&b)` fails to compile instead of auto-deref'ing to `(&a).min(&b)`, as it'd be an elegant way to use the `impl<T:Ord> Ord for &T` where borrow comparison is desirable...